src/main/java/seedu/address/model/order/DateChecker.java

```diff
@@ -103,6 +90,8 @@ public static String parsePotentialNaturalDate(String input) {
             String date = inputSplit[0];
             String time = inputSplit[1];
             dateToCheck = date;
+            timeToCheck = time;
+            isTimeBeforeCurrentTime = inputTimeBeforeCurrentTimeChecker();
             naturalDateCheck();
```

Comment:
violation of SLAP.
you might want to create another method to hide the details of the processing of the class variables before naturalDateCheck()

Reply:
Updated this. Good catch!